src/psyclone/transformations.py

```diff
@@ -1316,82 +1322,70 @@ def name(self):
+            raise TypeError("Expected nowait to be a bool "
+                            "but got {0}".format(value))
         self._omp_nowait = value
 
     def _directive(self, children):
         '''
         Creates the type of directive needed for this sub-class of
         transformation.
+
         :param children: list of Nodes that will be the children of \
                          the created directive.
         :type children: list of :py:class:`psyclone.psyir.nodes.Node`
-        :rtype: :py:class:`psyclone.psyGen.OMPDoDirective`
-        :raises NotImplementedError: if a collapse argument is supplied
-        '''
-        _directive = nodes.OMPSingleDirective(children=children,
-                                        nowait=self.omp_nowait)
-        return _directive
-
-    def validate(self, node_list, options=None):
-        '''
-        Perform OpenMP-specific validation checks.
 
-        :param node_list: list of Nodes to put within serial region.
-        :type node_list: list of :py:class:`psyclone.psyir.nodes.Node`
-        :param options: a dictionary with options for transformations.
-        :type options: dictionary of string:values or None
-        :param bool options["node-type-check"]: this flag controls if the \
-                type of the nodes enclosed in the region should be tested \
-                to avoid using unsupported nodes inside a region.
+        :returns: The directive created for the OpenMP Single Directive
+        :rtype: :py:class:`psyclone.psyGen.OMPSingleDirective`
 
-        :raises TransformationError: if the target Nodes are already within \
-                                     some OMP single region.
         '''
+        _directive = nodes.OMPSingleDirective(children=children,
+                                              nowait=self.omp_nowait)
+        return _directive
 
-        if node_list[0].ancestor(nodes.OMPSingleDirective):
-            raise TransformationError("Error in OMPSingle transformation:" +
-                                      " cannot create an OpenMP SINGLE " +
-                                      "region within another OpenMP SINGLE region.")
-
-        # Now call the general validation checks
-        super(OMPSingleTrans, self).validate(node_list, options)
-
-    def apply(self, node, options=None):
+    def apply(self, node_list, options=None):
         '''Apply the OMPSingleTrans transformation to the specified node in a
-        Schedule. 
+        Schedule.
 
         At code-generation time (when
-        :py:meth:`OMPLoopDirective.gen_code` is called), this node must be
-        within (i.e. a child of) an OpenMP PARALLEL region.
+        :py:meth:`OMPLoopDirective.gen_code` is called or when the PSyIR tree
+        is given to a backend), this node must be within (i.e. a child of) an
+        OpenMP PARALLEL region.
 
         If the keyword "nowait" is specified in the options, it will cause a
```

Comment:
This sentence bit a garbled is :-)

Reply:
I rewrote this, hopefully its better now.

Reply:
I think the last sentence needs to end "...otherwise no clause will be added."

Reply:
Almost - you've added it at line 451 when it should be at line 1365 or thereabouts.